PYIC-1270: Add validatecricheck lambda

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Add validatecricheck lambda 

### Why did it change

Add a lambda to validate the check returned by a CRI meets our needs or
reaching an M1A level of assurance.

If it doesn't meet the level required we return a "fail" journey
response. This will lead the user down a session end path (handled by
the journey engine).

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-1270](https://govukverify.atlassian.net/browse/PYIC-1270)

Wynndow committed:
BAU: Make ApiGatewayResponseGenerator ObjectMapper static

The ApiGatewayResponseGenerator was creating a new ObjectMapper for each
invocation. ObjectMappers are heavy objects and should be re-used.
Especially in lambda world.

Wynndow committed:
PYIC-1270: Add validatecricheck lambda

Add a lambda to validate the check returned by a CRI meets our needs or
reaching an M1A level of assurance.

If it doesn't meet the level required we return a "fail" journey
response. This will lead the user down a session end path (handled by
the journey engine).

dannyhayes-gds commented on deploy/template.yaml:
```diff
@@ -742,6 +742,54 @@ Resources:
+        - DynamoDBReadPolicy:
+            TableName: !Ref UserIssuedCredentialsTable
+      Events:
+        IPVCoreInternalAPI:
```
I think this has changed now. The internal gateway has changed to the private gateway. You might need to rebase off main to get those changes and see how this is now done

Wynndow replied:
Oh interesting - thanks i'll fix it up

Wynndow replied:
Done

Wynndow committed:
PYIC-1270: Use correct RestApiId

The resource name for the API gateway ID changed.

dannyhayes-gds commented on deploy/template.yaml:
```diff
@@ -766,6 +766,54 @@ Resources:
+        - DynamoDBReadPolicy:
+            TableName: !Ref UserIssuedCredentialsTable
+      Events:
+        IPVCoreInternalAPI:
```
Think this line needs changing as well?
```suggestion
        IPVCorePrivateAPI:
```

Wynndow replied:
Bah. Have fixed up. Thanks!
Just checking in with Dan about the open-api gateway spec. I can see the external one being referenced in the template, but not the internal one for the private API.

Wynndow committed:
PYIC-1270: Add validatecricheck to open-api spec

We're currently not using the open-api spec, but it makes sense to keep
it up to date in case we need it in future.
